manager/manager.go

```diff
@@ -0,0 +1,1210 @@
+	}
+	if id.UID == "" {
+		for _, probe := range m.Probes {
+			if probe.Section == id.Section {
+				programs = append(programs, probe.program)
+			}
+		}
+		if len(programs) > 0 {
+			return programs, true, nil
+		}
+		prog, ok := m.collection.Programs[id.Section]
+		return []*ebpf.Program{prog}, ok, nil
+	}
+	for _, probe := range m.Probes {
+		if probe.IdentificationPairMatches(id) {
+			return []*ebpf.Program{probe.program}, true, nil
+		}
+	}
+	return programs, false, nil
+}
+
+// GetProgramSpec - Return a pointer to the requested eBPF program spec
+// section: section of the program, as defined by its section SEC("[section]")
+// UID: unique identifier given to a probe. If UID is empty, then the original program spec with the right section in the
+// collection spec (if found) is return
+func (m *Manager) GetProgramSpec(id ProbeIdentificationPair) ([]*ebpf.ProgramSpec, bool, error) {
+	m.stateLock.RLock()
+	defer m.stateLock.RUnlock()
+
+	var programs []*ebpf.ProgramSpec
+	if m.collectionSpec == nil || m.state < initialized {
+		return nil, false, ErrManagerNotInitialized
+	}
+	if id.UID == "" {
+		for _, probe := range m.Probes {
+			if probe.Section == id.Section {
+				programs = append(programs, probe.programSpec)
+			}
+		}
+		if len(programs) > 0 {
+			return programs, true, nil
+		}
+		prog, ok := m.collectionSpec.Programs[id.Section]
+		return []*ebpf.ProgramSpec{prog}, ok, nil
+	}
+	for _, probe := range m.Probes {
+		if probe.IdentificationPairMatches(id) {
+			return []*ebpf.ProgramSpec{probe.programSpec}, true, nil
+		}
+	}
+	return programs, false, nil
+}
+
+// GetProbe - Select a probe by its section and UID
+func (m *Manager) GetProbe(id ProbeIdentificationPair) (*Probe, bool) {
+	for _, managerProbe := range m.Probes {
+		if managerProbe.IdentificationPairMatches(id) {
+			return managerProbe, true
+		}
+	}
+	return nil, false
+}
+
+// Init - Initialize the manager.
+// elf: reader containing the eBPF bytecode
+func (m *Manager) Init(elf io.ReaderAt) error {
+	return m.InitWithOptions(elf, Options{})
+}
+
+// InitWithOptions - Initialize the manager.
+// elf: reader containing the eBPF bytecode
+// options: options provided to the manager to configure its initialization
+func (m *Manager) InitWithOptions(elf io.ReaderAt, options Options) error {
+	if m.stateLock == nil {
+		m.stateLock = &sync.RWMutex{}
+	}
+	m.stateLock.Lock()
+	if m.state > initialized {
+		m.stateLock.Unlock()
+		return ErrManagerRunning
+	}
+
+	m.wg = &sync.WaitGroup{}
+	m.options = options
+	m.netlinkCache = make(map[netlinkCacheKey]*netlinkCacheValue)
+	if m.options.DefaultPerfRingBufferSize == 0 {
+		m.options.DefaultPerfRingBufferSize = os.Getpagesize()
+	}
+
+	// perform a quick sanity check on the provided probes and maps
+	if err := m.sanityCheck(); err != nil {
+		m.stateLock.Unlock()
+		return err
+	}
+
+	// set resource limit if requested
+	if m.options.RLimit != nil {
+		err := unix.Setrlimit(8, &unix.Rlimit{
```

Comment:
You can use the predefined constant here:
```suggestion
		err := unix.Setrlimit(unix.RLIMIT_MEMLOCK, &unix.Rlimit{
```

Reply:
Done !